### DISCUSSION TITLE
seisio.to_segy assertion error

### DISCUSSION THREAD

**Discussion by luke-harlowlawrence** [trhallam/segysak#131] [Help] [answered]
I am working on coblending 3 frequency segy volumes and then converting to greyscale for use in another program but am having problems writing the xarray dataset back to segy format. For this example, I have omitted my coblending script and am just attempting to read a segy file to xarray and then trying to write the xarray back to segy. The xarray works as intended, however when I try to convert back to segy I end up with an assertion error that is empty. 

```
#my working segy path
red_path = r"C:\Users\L\segy\SD_red.sgy"  

#load as xarray
try:
    red_volume = xr.open_dataset(
        red_path, 
        engine="sgy_engine", 
        dim_byte_fields={"iline": 189, "xline": 193} 
    )

except ValueError as e:
    print(f"Error loading SEGY file: {e}")

#output path
output_path = r"C:\Users\L\segy\red_volume_test.segy"

try:
    red_volume.seisio.to_segy('output_path', trace_header_map={'cdp_x': 73, 'cdp_y': 77}, iline=189, xline=193, vert_dimension='samples' )
    print(f"Test volume saved as SEG-Y file at {output_path}")
#except AssertionError as e:
    #print(f"AssertionError: {e}")
except Exception as e:
    print(f"An error occurred: {e}")
```

My only output is AssertionError: 
and no error to follow it. 

Any ideas? Thanks!

ps. here is my dataset info
```
RED <xarray.Dataset> Size: 122MB
Dimensions:  (iline: 351, xline: 288, samples: 301)
Coordinates:
  * iline    (iline) int16 702B 13 14 15 16 17 18 19 ... 358 359 360 361 362 363
  * xline    (xline) int16 576B 2 3 4 5 6 7 8 9 ... 283 284 285 286 287 288 289     
  * samples  (samples) float32 1kB 1.2e+03 1.202e+03 ... 1.798e+03 1.8e+03
Data variables:
    data     (iline, xline, samples) float32 122MB 0.0 0.0 0.0 ... 0.0 0.0 0.0      
Attributes:
    seisnc:   {"coord_scalar": -10.0, "coord_scaled": false}
```

**Comment by trhallam** [comment, marked as answer]
Two things.

Wrapping the function in a try-except block like this will hide the stack trace which is why the assertion error is not helpful. However, I will look to add more meaningful messages to the asset statements in the code.

Secondly, it looks like you are trying to write CDP coordinates, but have not loaded any into your dataset. My guess is that the assertion error is because the variables `cdp_x` and `cdp_y` don't exist in your dataset.

**Reply by lukelawrence123** [reply]
Hey thanks so much for the insight! I have a lot to learn still. It is working great now simply as:

red_volume.seisio.to_segy(output_path, iline=189, xline=193,vert_dimension='samples')
